### DISCUSSION TITLE
Debugger improvements

### DISCUSSION THREAD

**Discussion by holub** [mamedev/discussions#13] [Ideas]
We have very powerful debugger but it's mostly created for M.A.M.E. developers and very alien for developers who want to use it for development in target platform.

Some things which can be considered for improvements:

1. Run and start without breaking into debugger
2. Historical values for registers
3. Grouping view (e.g. for ports D-values)
4. Custom view for data region with defined names for offsets. Can be useful in REGs arrays.
5. Configurable key mappings

**Comment by Tafoid** [comment]
>     1. Run and start without breaking into debugger

This can be gotten around by using -debugscript and making a debug.txt file with "**g**" as the only line.
Then launch MAME with added:  **-debugscript debug.txt**

**Reply by holub** [reply]
That is awesome! Not too user-friendly though

**Comment by holub** [comment]
There are no custom hotkeys in dubugger or am I missing?

**Comment by holub** [comment]
5. Configurable mappings:
Custom window in debugger which allows to set up table with name and lua value.
I think that covers (4)

**Comment by AlKossow** [comment]
symbols?

**Comment by MooglyGuy** [comment]
All of these seem like great ideas, I eagerly await you implementing them.

**Comment by holub** [comment]
Given memory window has huge dropdown, it worth to keep only memory related items there and add another "state" view for the rest.
State items can be available on device level as a list + individual in the shape we currently have them.
